tests/functional/ProjectTask.php

```diff
@@ -401,4 +401,77 @@ public function testCloneProjectTask()
         $this->integer($clonedSubtask2->fields['projects_id'])->isEqualTo($project->getID());
         $this->string($clonedSubtask2->fields['name'])->isEqualTo($subtask2->fields['name'] . ' (copy)');
     }
+
+    public function testAutoSetDateForAdd()
+    {
+        $_SESSION['glpi_currenttime'] = '2023-10-10 10:10:10';
+        // Create a project
+        $project = $this->createItem('Project', [
+            'name' => 'Project 1',
+        ]);
+
+        // Create a project task with percent_done < 100 and > 0
+        $task = $this->createItem('ProjectTask', [
+            'projects_id' => $project->getID(),
+            'name'        => 'Task 1',
+            'percent_done' => '5',
+        ]);
+
+        // Check if the task has been added with the current date
+        $this->string($task->fields['real_start_date'])->isEqualTo(\Session::getCurrentTime());
+
+        // Create a project task with percent_done = 100
+        $task = $this->createItem('ProjectTask', [
+            'projects_id' => $project->getID(),
+            'name'        => 'Task 2',
+            'percent_done' => '100',
+        ]);
+
+        // Check if the task has been added with the current date
+        $this->string($task->fields['real_start_date'])->isEqualTo(\Session::getCurrentTime());
+        $this->string($task->fields['real_end_date'])->isEqualTo(\Session::getCurrentTime());
+    }
+
+    public function testAutoSetDateForUpdate()
```

Comment:
You should use a data provider for this test method, in order to be able to check the all possible case.
The data provider should adapt `$input` values and `$this->fields` values.

For instance, you should tests the behaviour of the method when empty strings are pass in `$input` (e.g. `$input['real_start_date'] = ''`). This is not a valid input, but as far as I see, nothing in the code would prevent such a value to be passed to the `prepareInputForAdd/Update()` method.